### PR TITLE
Clarify that beta annotations can not be removed

### PR DESCRIPTION
**Pull request by thockin**
Removing annotations caused a major outage with nodes that lagged the
master.  We used annotations as would-be fields, though we don't any
more.  Now we just add fields.  That makes them part of the API version.
These annotations are effectively part of the API version, too.

It stinks because I love to delete code, but I think it's just part of
the debt-accrual process and yet another reason we need to be judicious
in adding fields.

Allow edits from maintainers checkbox

### CODE REVIEW

**Comment by thockin** [comment]
@kubernetes/sig-architecture-pr-reviews

File: content/en/docs/reference/deprecation-policy.md [hunk 1]
```diff
@@ -36,6 +36,7 @@ includes:
 
    * REST resources (aka API objects)
    * Fields of REST resources
+   * Annotations on REST resources, including "beta" annotations
```
Comment (liggitt) [inline content/en/docs/reference/deprecation-policy.md]: What about alpha annotations?

Comment (liggitt) [reply content/en/docs/reference/deprecation-policy.md]: concretely, there's a current effort to remove/promote the alpha sysctl annotations to actual fields - https://github.com/kubernetes/kubernetes/pull/63717

**Comment by bgrant0607** [comment]
Additionally, the deprecation policy was inconsistent with the version skew policy:

https://github.com/kubernetes/community/blob/master/contributors/design-proposals/release/versioning.md#supported-releases-and-component-skew

Which doesn't take into account the 2 minor releases of skew between the master and nodes.

**Comment by bgrant0607** [comment]
And we need to ensure that any deprecations are pre-announced sufficiently far in advance, according to the policy. The annotation was removed without advance warning. This needs to be made clear in release-note guidelines. 

https://github.com/kubernetes/community/issues/484

File: content/en/docs/reference/deprecation-policy.md (hunk 1 above)
Comment (bgrant0607) [reply content/en/docs/reference/deprecation-policy.md]: Didn't we decide to break compatibility for alpha fields and annotations?

https://github.com/kubernetes/community/blob/master/contributors/devel/api_changes.md#adding-unstable-features-to-stable-versions

**Comment by bgrant0607** [comment]
BTW, github notifications are still mostly useless for me.

File: content/en/docs/reference/deprecation-policy.md (hunk 1 above)
Comment (liggitt) [reply content/en/docs/reference/deprecation-policy.md]: > Didn't we decide to break compatibility for alpha fields and annotations?

Yes, just wanted to make sure this wasn't worded in a way that seemed to prohibit that

Comment (thockin) [reply content/en/docs/reference/deprecation-policy.md]: Nope, I wanted to be clear that this is beta.  Do you want it to be more clear?

**Comment by thockin** [comment]
Re skew: Master on 1.10, nodes on 1.8 is the maximum skew.  So I will change the rules to 3 releases.

File: content/en/docs/reference/deprecation-policy.md (hunk 1 above)
Comment (liggitt) [reply content/en/docs/reference/deprecation-policy.md]: > Do you want it to be more clear?

I think so. Something like `Non-alpha annotations on REST resources (including "beta" annotations)`

**Comment by thockin** [comment]
New push is up.  PTAL especially at the table.

File: content/en/docs/reference/using-api/deprecation-policy.md [hunk 2]
```diff
@@ -75,15 +76,17 @@ versions.  Beta API versions *may not* replace GA API versions.
...
 
-   * **GA: 1 year or 2 releases (whichever is longer)**
-   * **Beta: 6 months or 2 releases (whichever is longer)**
+   * **GA: 12 months or 3 releases (whichever is longer)**
+   * **Beta: 9 months or 3 releases (whichever is longer)**
    * **Alpha: 0 releases**
 
+This covers the maximum supported version skew of 2 releases.
+
 NOTE: Until [#52185](https://github.com/kubernetes/kubernetes/issues/52185) is
```
Comment (liggitt) [inline content/en/docs/reference/using-api/deprecation-policy.md]: we should clarify this refers to being able to decode from storage, *not* that the older versions must continue to be served via the API. fine to do in a separate PR, just noticed it here in the diff.

File: content/en/docs/reference/using-api/deprecation-policy.md [hunk 3]
```diff
@@ -228,6 +231,12 @@ API versions are supported in a series of subsequent releases.
     </tr>
     <tr>
       <td>X+13</td>
+      <td>v2, v2beta1 (deprected), v2beta2 (deprecated), v1 (deprecated)</td>
```
Comment (liggitt) [inline content/en/docs/reference/using-api/deprecation-policy.md]: typo: deprected

**Comment by liggitt** [comment]
table looks good, just the typo and the clearer wording to exclude alpha annotations from the policy

File: content/en/docs/reference/using-api/deprecation-policy.md (hunk 3 above)
Comment (thockin) [reply content/en/docs/reference/using-api/deprecation-policy.md]: fixed in next push

**Comment by thockin** [comment]
New push is up

**Comment by liggitt** [comment]
Probably worth an email to kubernetes-dev as well. I mean, *I* review this doc on a daily basis, but I recognize not everyone may

**Comment by bgrant0607** [comment]
Sorry for the delay. LGTM also. Thanks for the updates.

We should get the version skew policy translated to a user-facing doc at some point.

File: content/en/docs/reference/using-api/deprecation-policy.md (hunk 2 above)
Comment (liggitt) [reply content/en/docs/reference/using-api/deprecation-policy.md]: follow-up in https://github.com/kubernetes/website/pull/8825
